src/pyramid/interfaces.py

```diff
@@ -483,33 +483,33 @@ def __call__(self, **kw):
 
 
 class ISecurityPolicy(Interface):
-    def identify(request):
-        """ Return an object identifying a trusted and verified user.  This
-        object may be anything, but should implement a ``__str__`` method that
-        outputs a corresponding :term:`userid`.
+    def authenticated_userid(request, identity):
+        """ Return a :term:`userid` string identifying the trusted and
+        verified user, or ``None`` if unauthenticated.
+        """
 
+    def identify(request):
+        """ Return the :term:`identity` of the current user.  The object can be
+        of any shape, such as a simple ID string or an ORM object.
         """
```

Comment:
Detail: this could be the first method defined in the interface.

Reply:
I wasn't sure which way to order them.  On one hand, suggested workflow is `identify` -> `authenticated_userid`, so by that lens `identify` comes first.  On the other hand, `authenticated_userid` is the "important" one, identity is just for convenience.

If you have a strong opinion I'm happy to reorder them.

Reply:
I don’t have an opinion yet; I’m still digesting the recent discussions (changes to the status and name of `identify` method).  I am installing from your branch in my project, let me see what it looks like now (I was using a dict for identity, so I will rewrite it now to rather rely on `unauthenticated_userid` in `permits`, makes more sense for this case!)